srtcore/core.cpp

```diff
@@ -12900,3 +13127,1190 @@ void CUDTGroup::synchronizeDrift(CUDT* cu, steady_clock::duration udrift, steady
         gi->ps->m_pUDT->m_pRcvBuffer->applyGroupDrift(newtimebase, wrap_period, udrift);
     }
 }
+
+
+// For sorting group members by priority
+
+struct FByPriotity //: public std::binary_predicate<CUDTGroup::gli_t, CUDTGroup::gli_t>
+{
+    typedef CUDTGroup::gli_t gli_t;
+    bool operator()(gli_t a, gli_t b)
+    {
+        // this should be operator <
+        return a->priority < b->priority;
+    }
+};
+
+int CUDTGroup::sendBackup(const char *buf, int len, SRT_MSGCTRL& w_mc)
```

Comment:
A function of one thousand lines! 🤣

Reply:
Sorry. There's really not much I can do about it. Maybe in future there could be done some refactoring to extract some common parts out of all sending functions in all groups, but we need to get all groups merged first.

The problem is that the algorithms and details of what is being done inside, were changing over time, including reinventing the algorithms from scratch. So I'd wait for the code to get worn more in use until we refax it.

Reply:
That is terrible. And this is not the only case. There was a plenty of code like this already merged and added to the technical debt. And now this 1k line function.
This can't be merged.